restcomm/restcomm.dao/src/main/java/org/restcomm/connect/dao/entities/Client.java

```diff
@@ -148,8 +149,14 @@ public Client setFriendlyName(final String friendlyName) {
                 voiceUrl, voiceMethod, voiceFallbackUrl, voiceFallbackMethod, voiceApplicationSid, uri, pushClientIdentity);
     }
 
-    public Client setPassword(final String password) {
-        return new Client(sid, dateCreated, DateTime.now(), accountSid, apiVersion, friendlyName, login, password, status,
+    /**
+     * @param login
+     * @param password
+     * @param realm
+     * @return
+     */
+    public Client setPassword(final String login, final String password, final String realm) {
+        return new Client(sid, dateCreated, DateTime.now(), accountSid, apiVersion, friendlyName, login, DigestAuthentication.HA1(login, realm, password), status,
```

Comment:
as mentionedin daily, i would prefer this kind of logic to be isolated in a Service class, and let the Entity as DTO container. The potential new service component could hold the logic of DigestAuthentication utility class as well. We may refactor this in a next iteration if scope in current Sprint is not appropiate.